evi-react-native-example/modules/audio/ios/SoundPlayer.swift

```diff
@@ -0,0 +1,102 @@
+                audioPlayer.volume = 1.0
+                
+                print("Starting chunk...")
+                let result = audioPlayer.play()
+                
+                // If you don't include the below lines, audio playback will happen at low volume.
+                // This is a workaround for what appears to be an odd bug in AVFoundation related
+                // to echo cancellation. See https://forums.developer.apple.com/forums/thread/721535
+                try AVAudioSession.sharedInstance().overrideOutputAudioPort(.none)
+                try AVAudioSession.sharedInstance().overrideOutputAudioPort(.speaker)
+
+
+                if (!result) {
+                    continuation.resume(throwing: NSError(domain: "AudioModule", code: 1, userInfo: [NSLocalizedDescriptionKey: "Could not play audio"]))
+                }
+            } catch {
+                continuation.resume(throwing: error)
+            }
+        }
+    }
+    
+    private func logAudioSessionInfo() {
+        let audioSession = AVAudioSession.sharedInstance()
+
+        // Log current route
+        print("Current Audio Route:")
+        let currentRoute = audioSession.currentRoute
+        for output in currentRoute.outputs {
+            print("Output - Port Type: \(output.portType.rawValue), Port Name: \(output.portName)")
+        }
+        for input in currentRoute.inputs {
+            print("Input - Port Type: \(input.portType.rawValue), Port Name: \(input.portName)")
+        }
+
+        // Log available inputs
+        print("\nAvailable Input Devices:")
+        for input in audioSession.availableInputs ?? [] {
+            print("Port Type: \(input.portType.rawValue), Port Name: \(input.portName)")
+        }
+
+        // Log output route from current session
+        print("\nAvailable Output Devices (Current Route):")
+        for output in currentRoute.outputs {
+            print("Port Type: \(output.portType.rawValue), Port Name: \(output.portName)")
+        }
+        
+        print("Audio session volume: \(audioSession.outputVolume)")
+        print("Audio player volume: \(lastAudioPlayer?.volume ?? 99999)")
+    }
+
+}
+
+// This class is constructed with a `continuation` as provided by `withUnsafeThrowingContinuation` and 
+// implements the `AVAudioPlayerDelegate` protocol so that the continuation will be resumed when the audio
+// has completed playing or encountered an error.
+class ContinuationDelegate : NSObject, AVAudioPlayerDelegate {
+    var continuation: UnsafeContinuation<Void, Error>?
+    init(continuation: UnsafeContinuation<Void, Error>) {
+        self.continuation = continuation
+    }
+    func audioPlayerDidFinishPlaying(_ player: AVAudioPlayer, successfully flag: Bool) {
+        print("Ending chunk...")
+        continuation?.resume(returning: ())
+    }
+    func audioPlayerDecodeErrorDidOccur(_ player: AVAudioPlayer, error: Error?) {
+        continuation?.resume(throwing: error!)
```

Comment:
this'll crash if you force-unwrap a nil error. may i suggest a `SoundPlayerError.unknownDecodingFailure` for the nil case and a `SoundPlayerError.decodingFailure(error)` when its non-nil

Reply:
This code changed a bit with [Enqueue audio on the swift side!](https://github.com/HumeAI/hume-api-examples/pull/100/commits/e8860647386f815409be0ff3f44c6c0feeadda93), but I introduced

```swift
public enum SoundPlayerError: Error {
    case invalidBase64String
    case couldNotPlayAudio
    case decodeError(details: String)
}
```
and changed the analogous code to

```swift
SoundPlayerError.decodeError(details: error?.localizedDescription ?? "Unknown error"))
```
which should prevent the crash